test/test.js

```diff
@@ -223,6 +223,42 @@ describe("datetime directive", function(){
 		
 		assert.equal(element.val(), "+0000");
 	});
+
+	it("datetime-timezone and custom timezone", function(){
+		$rootScope.date = new Date;
+		$rootScope.timezone = "'+0500'";
+
+		var element = $compile("<input type='text' datetime='Z' ng-model='date' datetime-timezone={{timezone}}>")($rootScope);
+
+		$rootScope.$digest();
+
+		assert.equal(element.val(), "+0500");
+	});
+
+	it("datetime-timezone and utc format with colon", function(){
+		$rootScope.date = new Date;
+		$rootScope.timezone = "'+07:00'";
+
+		var element = $compile("<input type='text' datetime='Z' ng-model='date' datetime-timezone={{timezone}}>")($rootScope);
+
+		$rootScope.$digest();
+
+		assert.equal(element.val(), "+0700");
+	});
+
+	it("dynamic datetime-timezone", function(){
+		var date = $rootScope.date = new Date;
+		$rootScope.timezone = "'+0500'";
+
+		var element = $compile("<input type='text' datetime='Z' ng-model='date' datetime-timezone={{timezone}}>")($rootScope);
+		$rootScope.$digest();
+		assert.equal(element.val(), $date(date, "Z", "+0500"));
+
+		$rootScope.timezone = "'+0800'";
+		$rootScope.$digest();
+		assert.equal(element.val(), $date(date, "Z", "+0800"));
```

Comment:
```  1) datetime directive dynamic datetime-timezone:

      AssertionError [ERR_ASSERTION]: '+0500' == '+0800'
      + expected - actual

      -+0500
      ++0800
```
@eight04 I have problem in this test, can you look and maybe help?

Reply:
Sure. Would you mind if I push directly to this branch?

Reply:
As I mentioned above, we have to `$watch` an expression or `attrs.$observe` the attribute. In this case, the attribute *does* change (you can inspect it through `element.attr("datetime-timezone")`) but the new value isn't used by the directive.

If the problem inside the `linkFunc` is fixed, the template of the dynamic test should be changed to:
```js
var element = $compile("<input type='text' datetime='Z' ng-model='date' datetime-timezone='timezone'>")($rootScope);
```

Reply:
@eight04 yes, you can push to this branch